cogni-libs/src/main/java/zone/cogni/libs/core/utils/ApacheHttpClientUtils.java

```diff
@@ -0,0 +1,109 @@
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility functions for working with the Apache HttpClient.
+ */
+public class ApacheHttpClientUtils {
+
+  private final static Logger log = LoggerFactory.getLogger(ApacheHttpClientUtils.class);
+
+  /**
+   * Builds a http client given username and password for authentication. Apache HttpClient is
+   * capable of delivering both Basic auth and Digest auth.
+   *
+   * @param username username to use for authentication
+   * @param password password to use for authentication
+   * @return http client
+   */
+  private static CloseableHttpClient buildHttpClient(final String username, final String password) {
+    final HttpClientBuilder httpClientBuilder = HttpClients.custom().useSystemProperties();
+    httpClientBuilder.setConnectionManager(
+        new PoolingHttpClientConnectionManager(60L, TimeUnit.SECONDS));
+
+    if (StringUtils.isNoneBlank(username, password)) {
+      CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+      credentialsProvider.setCredentials(AuthScope.ANY,
+          new UsernamePasswordCredentials(username, password));
+      httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
+    } else {
+      log.warn("Service is configured without credentials.");
+    }
+    return httpClientBuilder.build();
+  }
+
+  /**
+   * Ensures the response is 2xx. Throws a Runtime exception otherwise.
+   *
+   * @param response HTTP response to check.
+   */
+  private static void ensureResponseOK(HttpResponse response) {
+    if ((response.getStatusLine().getStatusCode() / 100) != 2) {
+      throw new RuntimeException(
+          "Not 2xx as answer: " + response.getStatusLine().getStatusCode() + " "
+              + response.getStatusLine().getReasonPhrase());
+    }
+  }
+
+
+  /**
+   * Executes and update request
+   *
+   * @param url          endpoint to reach
+   * @param username     to authenticate with
+   * @param password     to authenticate with
+   * @param addBasicAuth whether the "Authorization Basic ..." header shall be added
+   * @param httpEntity   payload
+   * @param put          whether a put (true) or a post (false)
+   * @param contentType  to send the data with
+   */
+  public static void executeAuthenticatedPostOrPut(
+      final String url,
+      final String username,
+      final String password,
+      final boolean addBasicAuth,
+      final HttpEntity httpEntity,
+      boolean put,
+      final String contentType) {
+
+    try (final CloseableHttpClient httpclient = ApacheHttpClientUtils.buildHttpClient(username,
+        password)) {
+      final HttpEntityEnclosingRequestBase httpPost =
+          put ? new HttpPut(url) : new HttpPost(url);
```

Comment:
:-) OK, I try to reformat through intellij, maybe I forgot, will check. Anyway, is there any automated formatter+checkstyle for the project? Given the size, it would pay off IMO, as these issues might be frequent ...

Reply:
@natan-cox-cognizone [here](https://github.com/cognizone/asquare/pull/229/commits/0548acadcef7411b67262f3dbd3ba2dc2a532799) you are.